metric-example/query-by-example.md

```diff
@@ -0,0 +1,516 @@
+```
+metricId,dt.entity.device_application.name,dt.entity.device_application,dt.entity.os.name,dt.entity.os,App Version,time,value  
+"builtin:apps.other.apdex.osAndVersion:names:filter(and(eq(dt.entity.os.name,iOS),prefix(App Version,6.)))",easyTravel Demo,MOBILE_APPLICATION-752C288D59734C79,iOS,OS-62028BEE737F03D4,6.8.7,1610993640000,0.87
+...
+```
+
+# Scenario 12: Using Space and Time Aggregation with Disk Usages
+
+>**Task**
+>For a dashboard, we want a single number for the used disk capacity over all hosts.
+
+`builtin:host.disk.used` captures used disk space and will give us the average per timeslot if no aggregation is specified. We can append `:max:splitBy():sum` to first extract the maximum for each time slot on every host and disk (`:max`, the time aggregation), then aggregate the maximum into a single data structure (`:splitBy()` removes all dimensions and merges all series), and extract the sum part from the merged data structure (`:sum`, the space aggregation). The result will be a sum of used bytes, considering the maximum for each time slot, over all hosts and all disks.
+
+This common transformation pattern looks like a mistake at first, because the output of `:max` is a plain number, which can not be aggregated with `:sum` (only `:value` is supported for plain numbers, which is a no-op) and `:splitBy` as we used it until now does not change the data type being passed through it. In our use case, `:splitBy` _does_ change the output data type, in order for `:sum` to become a supported aggregation type. This feature is often referred to as _bucket inference_, because `:splitBy` and `merge` intelligently decide which data structure (bucket) will be used to collect the merged values, by looking to the right for the next aggregation being applied to the merged values, changing the output data type of the merging.
+
+Conceptually, you can think of the next aggregation after a `:merge` or `:splitBy` to operate on a list of merged values, supporting any of min, max, avg, sum, median, percentile on that list. Depending on the space aggregation in use, the API will use a suitable data structure to model this conceptual list of merged values (typically a statistical summary or percentile estimator).
+
+# Scenario 13: Combine Series and Point Queries with Folding
+
+>**Task**
+>For a report, we want to query both per-month CPU usage, but also get an overall average.
+
+In some situations, it is useful to transmute a series result into a point result in any position of the transformer chain. One such example is when we want mixed series and point results.
+
+Say, we want to access `builtin:host.cpu.usage` over the complete last year from January to just before January this year (`now-y/y` to `now/y`). We want a resolution of `1M`, but we are also interested in the average value over the whole year. This can be done in a single bulk query, once using `:fold` and once omitting it:

```

Comment:
```suggestion
Say we want to access `builtin:host.cpu.usage` over the complete last year from January to just before January this year (`now-y/y` to `now/y`). We want a resolution of `1M`, but we are also interested in the average value over the whole year. This can be done in a single bulk query, once using `:fold` and once omitting it:

```